src/features/room/sagas/pin.saga.ts

```diff
@@ -26,6 +35,19 @@ function* pinRemote(owner: Address, roomId: RoomId, { streamrClient }: PinRemote
 
     const metadata = stream.extensions['thechat.eth']
 
+    const isTokenGated: boolean = yield isTokenGatedRoom(roomId, streamrClient)
+
+    if (isTokenGated && metadata.tokenAddress) {
```

Comment:
What's the logic here? What does a "token gated" room mean in a pin situation? I join if I meet the conditions?

Reply:
Exactly! From the perspective of a user who wants to join a token-gated room, pinning felt like the most natural action to attach it to.